parser/src/lib.rs

```diff
@@ -41,6 +40,24 @@ pub enum DisplayHint {
     Unknown(String),
 }
 
+impl FromStr for DisplayHint {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "µs" => DisplayHint::Microseconds,
+            "a" => DisplayHint::Ascii,
+            "b" => DisplayHint::Binary,
+            "x" => DisplayHint::Hexadecimal {
+                is_uppercase: false,
+            },
+            "X" => DisplayHint::Hexadecimal { is_uppercase: true },
+            "?" => DisplayHint::Debug,
+            _ => return Err("Error parsing `DisplayHint`".to_string()),
```

Comment:
Since we don't use this string for anything, might as well make it `()`.

Reply:
I didn't think about the option of `()` as an `type Err`. Good idea!